<!DOCTYPE html>
<html>
  <head>
    <title>cshl201811.utf8.md</title>
    <meta charset="utf-8">
    <link href="/libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="/libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="custom.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">




class: title-slide, center, middle
count: false

.banner[![](epiviz.png)]

.title[What comes next? Beyond integration of visual and computational interactive data analysis]

.author[HÃ©ctor Corrada Bravo]  
.other-info[@hcorrada]

.other-info[
Center for Bioinformatics and Computational Biology  
University of Maryland, College Park, USA  
CSHL Biological Data Science 2018, 08 November 2018
]

.logo[![](logo.png)]



---
layout: true

## A longitudinal study

---
class: split-50

.column[
- 30 subjects (8 F, 22M) challenged with enterotoxigenic _E. coli_
- 16S profiling of stool samples one day before, day of and 9 days after
- Diarrehal symptoms recorded
- Ciproflaxin (antibiotic) given after symptoms (or 9 days if no symptoms)
]

.column[
.center.image-50[![](img/popetal.png)]
]

.source[[Pop et al., 2016 BMC Genomics](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-2777-0)]

---

### Questions

.highlight[Is there an association between (pre-challenge) microbiome structure and response to challenge]

.highlight[Is there a time structure in the association between specific taxonomic units and presentation of symptoms]

---
layout: false

## Microbiome survey - data generation

.center.middle[![](img/pipeline.png)]

---
layout: true

## This talk

---

### **Metaviz**  

http://metaviz.org  
Interactive visualization methods for metagenomic data  
Features for longitudinal data in development  

---

### **`metagenomeSeq`**

http://bioconductor.org/packages/metagenomeSeq  
Statistical methods for metagenomic data analysis  
Including longitudinal data

---

### **`metavizr`**

http://bioconductor.org/packages/metavizr  
Integration between R/Bioconductor infrastructure and interactive visualization

---

.center.middle[.image-60[![](img/thistalk.png)]]



---
layout: false
class: inverse, center, middle

# Metaviz

.source[[Wagner et al., 2018 Nucleic Acids Research](https://academic.oup.com/nar/article/46/6/2777/4909991)]

---
class: split-50

## Genome Browsers

.column[

**Benefits**
- Perspective arises naturally to provide quick, intuitive navigation
- Genes and annotations integrated

**Track-based**

- Sequences and measurements are laid out across screen
]

.column[
.image-50[![](img/ucsc.png)]
]

.source[https://genome.ucsc.edu]

---
class: split-50

## Epiviz: Interactive Visualization for Functional Genomics

.column[
### Separate track-based and non track-based visualization

Data visualizations updated based on user interaction

### Integrated with R/Bioconductor
]

.column[
.center.image-40[![](img/epiviz.png)]
]

.source[[Chelaru et al. 2014, Nat. Methods]()]

---

## Microbiome data visualization challenges

### Features with a hierarchy
  - Need to view multiple levels of hierarchy made of thousands of nodes
  
### Need to support population surveys of many samples

### Support quantitative measurements with heatmaps, scatterplots, and boxplots

---

## Hierarchical navigation

Information Visualization technique for hierarchical data

Examine data at different levels and perform feature selection

### FacetZoom

.center[![](img/ipod.png)]

.source[Daschelt et al. 2008. CHI]

---
class: split-50

## Metaviz

.column[
Integrated, interactive taxonomic feature selection  
Statistically guided visual analysis  
Intuitive navigation for mechanism for hierarchical metagenomic data
]

.column[
.image-50[![](img/metaviz.png)]
]

&lt;!--demo: http://metaviz.cbcb.umd.edu/videos/msd16s.mp4--&gt;

.source[Wagner et al., 2018. Nucleic Acids Research]
---

## Metaviz

.middle.center.image[![](img/metaviz-msd16s.gif)]

---
class: split-50

## Integration with Bioconductor

.column[
Compute differential abundance with `metagenomeSeq`

Explore results through interactive visualization

]

.column[
.center.image-40[![](img/metavizr.png)]
]

&lt;!--demo: http://metaviz.cbcb.umd.edu/videos/metavizr.mp4--&gt;
.source[http://bioconductor.org/packages/metavizr]

---

## Integration with Bioconductor

.middle.center.image-70[![](img/metavizr.gif)]

---
exclude: true

## Metaviz design considerations

Goal is to visualize multiple samples at once

Storing count data directly in web page will render slowly when viewing many samples

Client-server allows aggregations and other statistical operations to be computed at server

---

## Architecture

.center.middle[.image-80[![](img/architecture.png)]]
---

## Deployment options

University of Maryland Metagenome Browser (14k metagenomes)    
http://metaviz.cbcb.umd.edu

`metavizr`  
http://bioconductor.org/packages/metavizr.html

Local install  
https://github.com/epiviz/Metaviz  
https://github.com/epiviz/metaviz-data-provider

---
layout: true


## Metaviz for longitudinal designs

---
class:split-40

.column[
  How can we effectively display longitudinal data characteristics  
  E.g., heatmap is not effective
]

.column[
.center.middle[.image-60[![](img/metaviz.png)]]
]

---
class:split-40

.column[
  How can we effectively display longitudinal data characteristics  
  
  Sparkline Matrix
]

.column[
.center.middle[.image-60[![](img/SparklineDemo.png)]]
]

---

&lt;!--.center.middle.image-70[![](img/SparklineDemo.png)]--&gt;
.center.middle.image-80[![](img/metaviz-longitudinal.gif)]

&lt;!--demo: http://metaviz-dev.cbcb.umd.edu/metaviz2/demo/sparkline.html--&gt;



---
layout: false
class: inverse, center, middle

# `metagenomeSeq`

.source[[Paulson et al., 2017. _biorxiv_](https://www.biorxiv.org/content/early/2017/01/10/099457)]  
.source[[Paulson et al., 2013. _Nat. Methods_](https://www.nature.com/articles/nmeth.2658)]

---
layout: true

## Statistical modeling

---
class: split-40

.column[
  Detect intervals of differential abundance between two groups of interest.
  ]

.column[
  .center.image-60[![](img/problem.png)]
  ]

---
  
  .center.middle.image-100[![](img/model.png)]

Fit using smoothing-spline ANOVA (`gss` package)

---
  
  ![](img/splineda.png)

---
layout: false
class: inverse, center, middle

# `metavizr`

.source[http://bioconductor.org/packages/metavizr]

---
layout: true

# Interactivity with spline model

---
class: split-40

.column[

```r
library(metavizr)
library(etec16s)
data(etec16s)

app &lt;- startMetaviz()
app$plot(etec16, type="TimeSeries", 
         formula=abundance~id+AntiGiven+
           time*AnyDayDiarrhea
)
```
]

.column[
.center.image-50[![](img/metaspline.png)]
]


---
class: split-40

.column[
Inference depends on model parameter (smoothness of function)  
  
Use interactivity to explore sensitivity  
]

.column[
.center.image-50[![](img/interspline.png)]
]

---
class: split-40

.column[
Inference depends on model parameter (smoothness of function)  
  
Use interactivity to explore sensitivity  
]

.column[
.center.image-60[![](img/splineset.png)]
]

---
class: split-40

.column[
Inference depends on model parameter (smoothness of function)  
  
Use interactivity to explore sensitivity  
  
High smoothness penalty
]

.column[
.center.image-50[![](img/verysmooth.png)]
]

---
class: split-40

.column[
Inference depends on model parameter (smoothness of function)  
  
Use interactivity to explore sensitivity  
  
Low smoothness penalty
]

.column[
.center.image-50[![](img/notsmooth.png)]
]



---
layout: true

## Statistically guided visualization

---
class: split-50

.column[
Let's revisit the Data Analysis modes

How can we make the connection between two modes of 
analysis tighter?
]

.column[
.center.image-50[![](img/tukey.png)]
]

---
class: split-50

.column[
One idea: statistically guided visual exploration via _proactive computation_

Back to Epiviz: epigenetic regulation of gene expression across multiple tumor types (Timp et al. 2016)
]

.column[
.center.image-50[![](img/feed.png)]
]

---
class: split-50

.column[
One idea: statistically guided visual exploration via _proactive computation_

Analysis runs in background (proactive computation)

"Interesting results" presented to user, guides exploration of data underlying result.
]

.column[
.center.image-50[![](img/feed.png)]
]

---

.center.image-90[![](img/epiviz-feed2.gif)]






---
layout: true

## Interactive Data Analysis

---
class: split-50

.column[
In general, how do we support interactive data analysis?

Workflow construction: keep pieces small, we like to do that for _reusability_, now let's do it for _data reflection_.

What is the "sufficient statistic" my tool produces that a user can interact with?
]

.column[
.center.image-40[![](img/tukey.png)]
]

---
class: split-40

.column[
One example: segment-based analysis of RNA-seq data

Transcript quantification as sufficient statistic? 
]

.column[
.center.image-40[![](img/yanagi_workflow.png)]
]

---
class: split-40

.column[
One example: segment-based analysis of RNA-seq data

**No**. There is (pseudo)-counting, and there is quantification.

Can we produce a "sufficient statistic" we can interact with _before_ quantification.
]

.column[
.center.image-40[![](img/yanagi_workflow.png)]
]

---
class: split-40

.column[
One example: segment-based analysis of RNA-seq data

Segment-counts as the "sufficient statistic"

.source[Gunady et al. Yanagi (2018) biorxiv]
]

.column[
.center.image-60[![](img/yanagi_vis.png)]
]

---

.center.image-90[![](img/yanagi.gif)]



---

layout: false

## Summary

### Exploration of hierarchical features (FacetZoom)

--

### Design for longitudinal experimental structure (SparklineMatrix)

--

### Interact with data and statistical model via visualization (`metavizr`)

--

### Design your tools with data reflection, build IDA tools around that

---
layout: false
class: split-60

## Acknowledgements

.column[
### **CBCB/UMD College Park**

Justin Wagner, Joseph Paulson, Jayaram Kancherla, Florin Chelaru  
Mihai Pop, Niklas Elmqvist, Zhe Cui, Mohamed Gunady, Stephen Mount

### **UMD Baltimore**

Brianna Lindsey, O. Colin Stine, Owen White, Anup Mahurkar
]

.column[
### **Funding**

NIH/NIGMS, Genentech

### **More info**
http://epiviz.org
http://www.hcbravo.org
]
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>
<script>
remark.macros['scale'] = function (percentage) {
  var url = this;
  return '<img src="' + url + '" style=width: ' + percentage + '"/>';
};
</script>

<script>
(function() {
  var i, text, code, codes = document.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
})();
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
